tests/test_json_to_parquet.py

```diff
@@ -0,0 +1,531 @@
+import json
+import os
+import time
+import boto3
+import pandas
+import pytest
+from awsglue.context import GlueContext
+from pyspark.sql.session import SparkSession
+from src.glue.jobs import json_to_parquet
+from unittest.mock import patch
+# requires pytest-datadir to be installed
+
+
+@patch(
+        "src.glue.jobs.json_to_parquet.INDEX_FIELD_MAP",
+        {"TestFlatDataType": ["GlobalKey"], "TestNestedDataType": ["GlobalKey"]}
+)
+class TestJsonS3ToParquet:
+
+    @pytest.fixture(scope="class")
+    def glue_database_name(self, namespace):
+        return f"{namespace}-pytest-database"
```

Comment:
Nit: Can these be `yield` instead of `return`?

Reply:
They can be, but why would I want to use yield?

Reply:
`yield` can be used as a way to "setup" and "teardown" resources.  Just a minor thing.